### PR TITLE
Fixes #4420 - ORM belongs_to relationships can now be set to NULL

### PR DESCRIPTION
**Pull request by ghost**
The foreign key in the belongs_to relationship is now only set to the value's primary key if the value is an instance of ORM, otherwise the foreign key is set to NULL.

### CODE REVIEW

File: classes/kohana/orm.php [hunk 1]
```diff
@@ -678,7 +678,10 @@ public function set($column, $value)
 			$this->_related[$column] = $value;
 
 			// Update the foreign key of this model
-			$this->_object[$this->_belongs_to[$column]['foreign_key']] = $value->pk();
+			$this->_object[$this->_belongs_to[$column]['foreign_key']]
+				= ($value instanceof ORM)
```
Comment (zeelot) [inline classes/kohana/orm.php]: Could you change this line to match our coding style? http://kohanaframework.org/3.2/guide/kohana/conventions#ternaries

**Comment by radsystems** [comment]
Do you mean by having the assignment operator on the same line? I split it into multiple lines as it went beyond 80 chars as per the coding style.

**Comment by zeelot** [comment]
I see your point, but think it's more important to format the ternary properly than it is to keep the line at or below 80 characters.

What you could do, if you wanted to follow both conventions strictly, is split the ternary into its own line.

```
    elseif (isset($this->_belongs_to[$column]))
    {
        // Update related object itself
        $this->_related[$column] = $value;

        // Update the foreign key of this model
        $new_fk = ($value instanceof ORM)
            ? $value->pk()
            : NULL;

        $this->_object[$this->_belongs_to[$column]['foreign_key']] = $new_fk;

        $this->_changed[$column] = $this->_belongs_to[$column]['foreign_key'];
    }
```

Generally though, I don't think it's a big deal to go a little beyond 80 on a line. But that might just be me...

**Comment by radsystems** [comment]
The 80 char limit seems unnecessary, so I've put the assignment on the same line.

**Pull request merged**